proto/viam/service/mlmodel/v1/mlmodel.proto

```diff
@@ -91,3 +98,70 @@ enum LabelType {
   // the position of the tensor value in the axis is the label index
   LABEL_TYPE_TENSOR_AXIS = 2;
 }
+
+message FlatTensorDataInt8 {
+  bytes data = 1;
+}
+
+message FlatTensorDataUInt8 {
+  bytes data = 1;
+}
+
+message FlatTensorDataInt16 {
+  // Pack two 16-bit numbers per entry.
+  // TODO: Describe handling for BE
+  repeated fixed32 data = 1 [packed = true];
+}
+
+message FlatTensorDataUInt16 {
+  // Pack two 16-bit numbers per entry.
+  // TODO: Describe handling for BE
+  repeated fixed32 data = 1 [packed = true];
+}
+
+message FlatTensorDataInt32 {
+  repeated sfixed32 data = 1 [packed = true];
+}
+
+message FlatTensorDataUInt32 {
+  repeated fixed32 data = 1 [packed = true];
+}
+
+message FlatTensorDataInt64 {
+  repeated sfixed64 data = 1 [packed = true];
+}
+
+message FlatTensorDataUInt64 {
+  repeated fixed64 data = 1 [packed = true];
+}
+
+message FlatTensorDataFloat {
+  repeated float data = 1 [packed = true];
+}
+
+message FlatTensorDataDouble {
+  repeated double data = 1 [packed = true];
+}
+
+message FlatTensor {
+  // TODO
+  repeated fixed64 shape = 1;
```

Comment:
We could potentially use a smaller type, but I don't think it matters. I don't expect shape vectors to be more than a handful of elements long, so any savings is pretty small. I sort of like using fixed64 since it maps 1:1 with `size_t` on the C++ side.

Reply:
@kharijarrett - Are you good with the decision to keep these as 64 bits?